### PR TITLE
[PM-8807] migrate desktop to generator libraries

### PR DESCRIPTION
**Pull request by audrey-jensen**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-8807

## 📔 Objective

Replace dependencies on `@bitwarden/common/tools/generator` with `@bitwarden/generator-core` and `@bitwarden/generator-extensions` so that the common code can be deleted.

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by ike-kottlowski — DISMISSED** [review]

**Review by djsmith85 — APPROVED** [review]

**Review by djsmith85 — CHANGES_REQUESTED** [review]
All reviewed changes are looking good, just seen that one test is failing

File: apps/desktop/src/app/tools/generator.component.spec.ts [hunk 1]
```diff
@@ -29,11 +31,11 @@ describe("GeneratorComponent", () => {
       providers: [
         {
           provide: PasswordGenerationServiceAbstraction,
-          useValue: mock<PasswordGenerationServiceAbstraction>(),
+          useValue: mock<typeof PasswordGenerationServiceAbstraction>(),
```
Comment (djsmith85) [inline apps/desktop/src/app/tools/generator.component.spec.ts:34]: ❓ Do we still need typeof here as it's now directly imported?

**Review by djsmith85 — DISMISSED** [review]
Changes are looking good, just one small conflict likely introduced by https://github.com/bitwarden/clients/pull/9690

File: apps/desktop/src/app/tools/generator.component.spec.ts (hunk 1 above)
Comment (audrey-jensen) [reply apps/desktop/src/app/tools/generator.component.spec.ts]: Resovled in b5e317ab.

**Review by djsmith85 — APPROVED** [review]

**Review by ike-kottlowski — APPROVED** [review]

**Pull request closed without merging**